### ISSUE TITLE
Training on TPU Pod

### ISSUE THREAD

**Issue by NaxAlpha** [google-research/text-to-text-transfer-transformer#127]
Hi,

I have been trying to fine-tune T5 on TPU Pod (v3-256). I changed TPU_TOPOLOGY to 8x16 and (3B) model parallelism is 8 and batch size 64. During training, it gets stuck after a few iterations:

![image](https://user-images.githubusercontent.com/11090613/76493249-06a92400-6454-11ea-91de-e4a3a262b260.png)

Can you help me what i am doing wrong here?

Thanks

**Comment by craffel** [comment]
Sorry, I've never seen this. Let me reach out to the Cloud TPU people to see if they can help debug.

**Comment by adarob** [comment]
I do recall I had to use "16x8" (vs "8x16") when training this internally,
but that could be unrelated here...

**Comment by NaxAlpha** [comment]
For more context, i am using TPU TF version 1.15. Rest is same. it works on single TPUv3 but I do not know what is causing the issue. More experiments yield that it occurs after a fixed number of iterations e.g. for above mentioned config, it gets stuck after 2200 iterations.

**Comment by craffel** [comment]
If it gets stuck and you kill the process and re-start, does it pick back up from where it left of and continue for 2200 iterations before stopping again?

**Comment by NaxAlpha** [comment]
yeah it works that way. So I have changed the script so that fine tune steps are at 2000 increments and after 2000 iterations, it restarts training for 2000 more steps. But TPU model loading and the whole bootstrapping process is so slow that i am getting half the throughput (like 20 minute training and 20 minute bootstrapping) than if it did not get stuck. Just wanted to ask if you guys faced similar issue.

**Comment by craffel** [comment]
The only place we have seen a similar issue is if we ran eval on the model on too much data. I've never seen it during any fine-tuning or training. I've reached out to some Cloud TPU people who can hopefully look into it.
